### PR TITLE
Moving body weight history and pen history into separate files 

### PR DESCRIPTION
**Pull request by ltp23**
This PR is pretty simple. I moved the `BodyWeightHistory` and `PenHistory` classes into their separate files and wrote unit tests for them. I also added more type annotations for the  `AnimalEvents` class.

### CODE REVIEW

**Review by JoeWaddell — APPROVED** [review]
Always nice to see more documentation and testing. Clearly written docstrings, runs on my system, and all tests pass. Looks good to me!

File: RUFAS/routines/animal/life_cycle/animal_events.py [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
 """
 
 
-class AnimalEvents(object):
-    def __init__(self):
+class AnimalEvents:
+    """
+    A class to represent animal events in a farm simulation.
+
+    This class tracks the events related to animals, including birth and other significant
```
Comment (ew3361zh) [inline RUFAS/routines/animal/life_cycle/animal_events.py]:
```suggestion
    This class tracks the events related to animals including birth and other significant
```

File: RUFAS/routines/animal/life_cycle/animal_events.py [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
+    occurrences.
+
+    Attributes
+    ----------
+    events : dict[int, list[str]]
+        A dictionary containing the events, indexed by the animal's age in days. The values
```
Comment (ew3361zh) [inline RUFAS/routines/animal/life_cycle/animal_events.py]:
```suggestion
        A dictionary containing the events indexed by the animal's age in days. The values
```

File: RUFAS/routines/animal/life_cycle/pen_history.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+class PenHistory:
+    """
+    A class to represent the history of a pen on a farm.
+
+    This class tracks the usage of a pen, including the start and end dates, the specific pen, and the classes
```
Comment (ew3361zh) [inline RUFAS/routines/animal/life_cycle/pen_history.py]:
```suggestion
    This class tracks the usage of a pen including the start and end dates, the specific pen, and the classes
```

**Review by ew3361zh — APPROVED** [review]
Looks great, Loi! A few miniscule suggested changes for (imo) unnecessary commas.

**Comment by ltp23** [comment]
Thanks Joe and Niko!

**Pull request merged**